dclab/rtdc_dataset/feat_temp.py

```diff
@@ -70,12 +72,17 @@ def set_temporary_feature(rtdc_ds, feature, data):
     if not feat_logic.feature_exists(feature):
         raise ValueError(
             f"Temporary feature '{feature}' has not been registered!")
-    if isinstance(rtdc_ds, RTDC_Hierarchy):
-        raise NotImplementedError("Setting temporary features for hierarchy "
-                                  "children not implemented yet!")
     if len(data) != len(rtdc_ds):
         raise ValueError(f"The temporary feature {feature} must have same "
                          f"length as the dataset. Expected length "
                          f"{len(rtdc_ds)}, got length {len(data)}!")
+    if isinstance(rtdc_ds, RTDC_Hierarchy):
+        root_ids = map_indices_child2root(rtdc_ds, np.arange(len(rtdc_ds)))
+        root_parent = rtdc_ds.get_root_parent()
+        root_feat_data = np.empty((len(root_parent)))
+        root_feat_data[:] = np.nan
+        root_feat_data[root_ids] = data
+        feat_logic.check_feature_shape(feature, data)
```

Comment:
I think it would be better to do

```
set_temporary_feature(root_parent, feature, root_feature_data)
```

This way you avoid having duplicate code which is good in case things change in the future.

Reply:
Good point!